static_compress/mixin.py

```diff
@@ -71,47 +72,64 @@ def get_modified_time(self, name):
         return self._datetime_from_timestamp(getmtime(alt))
 
     def post_process(self, paths, dry_run=False, **options):
-        if hasattr(super(), "post_process"):
-            yield from super().post_process(paths, dry_run, **options)
-
         if dry_run:
             return
 
+        files = OrderedDict()
+
+        super_class = super()
+        if hasattr(super_class, "post_process"):
+            generator = super_class.post_process(paths, dry_run, **options)
+        else:
+            generator = iter((f, f, False) for f in paths)
+
+        for file, hashed_file, processed in generator:
+            files[file] = hashed_file
+            yield file, hashed_file, processed
+
         for name in paths.keys():
-            if not self._is_file_allowed(name):
+            if not (name in files and self._is_file_allowed(name)):
                 continue
 
-            source_storage, path = paths[name]
+            source_storage, _ = paths[name]
+
+            path = files[name]
             # Process if file is big enough
-            if os.path.getsize(self.path(path)) < self.minimum_kb * 1024:
+            real_path = self.path(path)
+            if self.minimum_kb and os.path.getsize(real_path) < self.minimum_kb * 1024:
```

Comment:
I don't think this.minimum_kb check needs to be implemented? It is always set in init with default value

Reply:
you do not need to execute os.path.getsize if minimum_kb == 0